From b41d5c11f626c00bafc226db4d5a72b30cad01d4 Mon Sep 17 00:00:00 2001
From: justforlxz <justforlxz@gmail.com>
Date: Thu, 18 Mar 2021 12:43:50 +0800
Subject: [PATCH] refactor: Only enable dxcb in dde

The default use of dxcb may affect other programs

Issue: Fixes https://github.com/linuxdeepin/qt5integration/issues/41
Issue: Fixes https://github.com/keepassxreboot/keepassxc/issues/6277
Log:
Change-Id: I465e98128c8374b43197377f10a299601d3618b4
---
 xcb/main.cpp | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/xcb/main.cpp b/xcb/main.cpp
index ee8ded4..ceb5fd4 100644
--- a/xcb/main.cpp
+++ b/xcb/main.cpp
@@ -36,16 +36,21 @@ class DPlatformIntegrationPlugin : public QPlatformIntegrationPlugin
 QPlatformIntegration* DPlatformIntegrationPlugin::create(const QString& system, const QStringList& parameters, int &argc, char **argv)
 {
 #ifdef Q_OS_LINUX
+    bool loadDXcb = false;
+
     if (qEnvironmentVariableIsSet("D_DXCB_DISABLE")) {
-        return new DPlatformIntegrationParent(parameters, argc, argv);
+        loadDXcb = false;
+    } else if (system == "dxcb") {
+        loadDXcb = true;
+    } else if (QString(qgetenv("XDG_CURRENT_DESKTOP")).toLower().startsWith("deepin")) {
+        loadDXcb = true;
     }
 
-    if (system == "dxcb" || system == "xcb") {
-        return new DPlatformIntegration(parameters, argc, argv);
-    }
+    return loadDXcb ? new DPlatformIntegration(parameters, argc, argv)
+                    : new DPlatformIntegrationParent(parameters, argc, argv);
 #endif
 
-    return 0;
+    return nullptr;
 }
 
 QT_END_NAMESPACE
